src/units/unit.cpp

```diff
@@ -1408,6 +1408,181 @@ void unit::remove_ability_by_id(const std::string& ability)
 	}
 }
 
+static bool bool_matches_if_present(const config& filter, const config& cfg, const std::string& attribute, bool def)
+{
+	if(filter[attribute].empty()) {
+		return true;
+	}
+
+	return filter[attribute].to_bool() == cfg[attribute].to_bool(def);
+}
+
+static bool string_matches_if_present(const config& filter, const config& cfg, const std::string& attribute, const std::string& def)
+{
+	if(filter[attribute].empty()) {
+		return true;
+	}
+
+	const std::vector<std::string> filter_attribute = utils::split(filter[attribute]);
+	return ( std::find(filter_attribute.begin(), filter_attribute.end(), cfg[attribute].str(def)) != filter_attribute.end() );
+}
+
+static bool double_matches_if_present(const config& filter, const config& cfg, const std::string& attribute, const double& def)
+{
+	if(filter[attribute].empty()) {
+		return true;
+	}
+
+	return filter[attribute].to_double() == cfg[attribute].to_double(def);
```

Comment:
Floating-point numbers are a little inaccurate, so shouldn't be compared with the `==` operator.

Oddly enough, we don't seem to have a utility function for this in src/utils/math.h. However, that file does already have `in_ranges`, which might be the right function to use combined with the range-parsing functions.

Reply:
for int, yes but not for double

Reply:
https://github.com/wesnoth/wesnoth/pull/7168/commits/15907b7f2141b16b529559d8fa17413d214cfb7a must be reponseto issue

Reply:
It's unfortunate that `in_ranges` supports double but `parse_ranges` does not, so I added 73a1b46bbfeecf24e5fb3c4b95438ade1f7a7064. Does that help?

Reply:
@CelticMinstrel Yes, the compilation works fine, and frankly, it bothers me to have to specify every value even in a list instead of defining a range within which any included value matches. Thanks @CelticMinstrel for your help.

Reply:
i pushed changes in PR